contrib/util/dupscore.cli.php

```diff
@@ -0,0 +1,155 @@
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Rod Roark <rod@sunsetsystems.com>
+ * @copyright Copyright (c) 2021 Rod Roark <rod@sunsetsystems.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+// The number of scores to compute between tests for time expiration.
+$querylimit = 1000;
+
+function sqlExec($link, $query)
+{
+    if (!mysqli_query($link, $query)) {
+        die("Query failed: $query\n");
+    }
+    return true;
+}
+
+function sqlSelect($link, $query)
+{
+    $res = mysqli_query($link, $query);
+    if (!$res) {
+        die("Query failed: $query\n");
+    }
+    return $res;
+}
+
+function sqlSelectOne($link, $query)
+{
+    $res = sqlSelect($link, $query);
+    $row = mysqli_fetch_assoc($res);
+    mysqli_free_result($res);
+    return $row;
+}
+
+$args = getopt('cq', array('webdir:', 'site:', 'maxmins:'));
+
+// print_r($args); // debugging
+
+$args['webdir'] = $args['webdir'] ?? dirname(dirname(dirname(__FILE__)));
+$args['site'] = $args['site'] ?? 'default';
+$args['maxmins'] = floatval($args['maxmins'] ?? 60);
+
+if (stripos(PHP_OS, 'WIN') === 0) {
+    $args['webdir'] = str_replace("\\", "/", $args['webdir']);
+}
+
+$confname = $args['webdir'] . "/sites/" . $args['site'] . "/sqlconf.php";
+if (!is_file($confname)) {
+    die("File not found: " . $confname . "\n");
+}
+
+// Bring in the getDupScoreSQL() function.
+require_once($args['webdir'] . "/library/dupscore.inc.php");
+
+$link = false;
+include($confname);
+$link = mysqli_connect($host, $login, $pass, $dbase, $port);
+if (empty($link)) {
+    die("Failed to open database '$dbase'\n");
+}
+
+if (!isset($args['q'])) {
+    echo "Opened database '$dbase'.\n";
+}
+
+sqlExec($link, "SET sql_mode = ''");
+
+$endtime = time() + 365 * 24 * 60 * 60; // a year from now
+if (!empty($args['maxmins'])) {
+    $endtime = time() + $args['maxmins'] * 60;
+}
+
+if (isset($args['c'])) {
+    // Note -1 means the patient is manually flagged as not a duplicate.
+    sqlExec($link, "UPDATE patient_data SET dupscore = -9 WHERE dupscore != -1");
+    if (!isset($args['q'])) {
+        echo "All scores have been cleared.\n";
+    }
+}
+
+$count = 0;
+$finished = false;
+
+while (!$finished && time() < $endtime) {
+    $scores = array();
+    $query1 = "SELECT p1.pid, MAX(" . getDupScoreSQL() . ") AS dupscore" .
+        " FROM patient_data AS p1, patient_data AS p2" .
+        " WHERE p1.dupscore = -9 AND p2.pid < p1.pid" .
+        " GROUP BY p1.pid ORDER BY p1.pid LIMIT $querylimit";
+
+    // echo "$query1\n"; // debugging
+
+    $res1 = sqlSelect($link, $query1);
+    while ($row1 = mysqli_fetch_assoc($res1)) {
+        $scores[$row1['pid']] = $row1['dupscore'];
+    };
+
+    mysqli_free_result($res1);
+    foreach ($scores as $pid => $score) {
+        sqlExec($link, "UPDATE patient_data SET dupscore = $score WHERE pid = $pid");
```

Comment:
would pass the variable through the sql escaping function with quotes around them (mainly to ensure any expansion of functionality in this script will then follow this)

Reply:
Will use binding.